bowling/include/Game.hpp

```diff
@@ -0,0 +1,27 @@
+#pragma once
+
+#include <fstream>
+#include <iostream>
+#include <vector>
+
+#include "BowlingLane.hpp"
+#include "DataParser.hpp"
+#include "DirectoryHandler.hpp"
+
+class Game {
+public:
+    Game(const std::string& directoryPath, const std::string& resultOutput = "")
+        : directoryHandler_(directoryPath), resultOutput_(resultOutput) {
+        processData();
+    }
+
+    std::vector<std::shared_ptr<BowlingLane>> getLanes() const { return lanes_; }
+    void printResult();
+    void saveDataToFile();
```

Comment:
how to test whole game results?

you should have a function returning a string that is later either displayed on-screen or saved into file.
Maybe processData should return it?

Reply:
For this i created function getGameResult() which is creating a string of game result output, putting that functionality into processData will be against single function responsibility(processData is responsible for filling a game data from clean string input)